prosody-plugins/mod_auth_token.lua

```diff
@@ -64,32 +77,70 @@ function provider.delete_user(username)
 	return nil;
 end
 
+local http_timeout = 30;
+local http_headers = {
+	["User-Agent"] = "Prosody ("..prosody.version.."; "..prosody.platform..")"
+};
+
+-- TODO: This *needs* to be memoized before going to prod.
+function get_public_key(keyId)
+	local wait, done = async.waiter();
+	local content, code; --, request, response;
+	local function cb(content_, code_, response_, request_)
+		content, code = content_, code_;
+		done();
+	end
+	local request = http.request(path.join(asapKeyServer, keyId), {
+		headers = http_headers or {},
+		method = "GET"
+	}, cb);
+	-- TODO: Is the done() call racey?
+	timer.add_task(http_timeout, function() http.destroy_request(request); done(); end);
+	wait();
+
+	if code == 200 or code == 204 then
+		return content;
+	end
+
+	return nil
+end
+
 function provider.get_sasl_handler(session)
 	-- JWT token extracted from BOSH URL
 	local token = session.auth_token;
 
 	local function get_username_from_token(self, message)
 
 		if token == nil then
-			if allowEmptyToken == true then
-				return true;
+			if allowEmptyToken then
```

Comment:
Is it necessary for some reason? I don't think anything was wrong with it, it just looked odd. I don't really know Lua, so if there was a reason for this I'm happy to change it back.